bodhi/models/models.py

```diff
@@ -1077,6 +1077,16 @@ def set_request(self, db, action, username):
                 update=self, agent=username))
             return
 
+        # If an update with pending status receives negative karma, make sure it gets obsoleted
+        if self.status is UpdateStatus.pending and self.request is UpdateRequest.testing \
+            and self.karma < 0:
```

Comment:
@bowlofeggs Yeah I agree with autokarma turned on.
Okay one more query regarding negative karma. Do we want to check negative karma from recent comment like we did `has_negative_karma` or just checking `unstable_karma(negative_karma) threshold` would be choice here?